docs/design/operator.md

```diff
@@ -0,0 +1,200 @@
+
+#### Storage Capacity
+
+The Gluster operator will monitor the storage capacity of the overall cluster.
+Storage capacity is defined here as the amount of space that can be committed to
+new Gluster volumes and not how much free space is available. It will trigger
+events when storage capacity reaches "Warning" levels and "Critical" levels.
+These levels will be configurable, and should be set to a value that will allow
+for corrective action to take place before the storage reaches maximum capacity.
+
+**Target Configurations:** External, Hosted, Managed
+
+#### Volume Density
+
+The Gluster operator will monitor the total volume density of any cluster it is
+automating. It will trigger events when the number of volumes in the cluster
+reaches a pre-defined level that should prompt corrective action. This value
+will be configurable, and should be set to a value that will allow for
+corrective action to take place before cluster nodes lag or crash.
+
+**Target Configurations:** External, Hosted, Managed
+
+#### Device Health
+
+The Gluster operator will monitor the health of all devices in an automated
+Gluster cluster. It will report a "Warning" when a device reports failing health
+and will report a "Critical" problem when a device becomes inaccessible. Support
+for this feature may vary form one device type to another.
+
+**Target Configurations:** External, Hosted, Managed
+
+#### Volume Health
+
+The Gluster operator will monitor the health of all volumes in an automated
+Gluster cluster. It will report a "Warning" when a volume reports failing bricks
+and will report a "Critical" problem when a volume becomes inaccessible.
+
+**Target Configurations:** External, Hosted, Managed
+
+#### Node Health
+
+The Gluster operator will monitor the health of all nodes in an automated
+Gluster cluster. It will report a "Critical" problem when a node becomes
+inaccessible.
+
+**Target Configurations:** External, Hosted, Managed
+
+### Cluster Management
+
+In response to the monitoring events above, the Gluster operator will take any
+number of corrective management actions.
+
+#### Storage Scaling
+
+In the event that the overall cluster reports a critical capacity level, the
+operator will deploy a set number of additional storage nodes with new storage
+devices attached. The number of additional nodes to be deployed will be
+configurable.
+
+**Target Configurations:** Managed
+
+#### Cluster Scaling
+
+In the event that the overall cluster reports a critical volume density level,
+the operator will deploy a new cluster of Gluster nodes to receive volume
+creation requests. The number of additional nodes to be deployed will be
+configurable.
+
+**Target Configurations:** Managed
+
+### Disaster Recovery
+
+In response to the health monitoring events above, the Gluster operator will
+take any number of corrective actions to recover from a situation that could
+result in potential data loss.
+
+#### Brick Replacement
```

Comment:
Per the meeting last month, this will be a Gluster-internal operation, assuming sufficient free capacity is available. The operator is only concerned with adding and removing capacity & nodes from the cluster.

Reply:
Oh, right! Fair enough. Tying this back to the monitoring section, would it make sense to have some sort of "out of space" event that the operator could react to in addition to the two watermarks?